src/main/java/spoon/support/sniper/SniperJavaPrettyPrinter.java

```diff
@@ -221,7 +221,6 @@ public void onTokenWriterWrite(TokenType tokenType, String token, CtComment comm
 			public void printSourceFragment(SourceFragment fragment, ModificationStatus isModified) {
 				if (isModified == ModificationStatus.UNKNOWN || isModified == ModificationStatus.MODIFIED) {
 					printAction.run();
-					return;
 				} else {
 					if (fragment instanceof CollectionSourceFragment) {
 						//we started scanning of collection of elements
```

Comment:
IMO this thing is much worse than that early return

```suggestion
				} else if (fragment instanceof CollectionSourceFragment) {
						//we started scanning of collection of elements
```